### DISCUSSION TITLE
Smart Contract Sharing Guide

### DISCUSSION THREAD

**Discussion by parallelchainlab** [parallelchain-io/pchain-sdk#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

## Smart Contract Sharing
Feel free to share the contract address of your smart contract, along with its arguments. We would like everyone to make calls to different smart contracts.

**Comment by mentasuave01** [comment]
Hi I'm having problems with build command: cargo build --target wasm32-unknown-unknown --release
Build always fail I have pchain, rust and wasm well installed.
I have the same error on default example contracts.

Example of error:

`error[E0425]: cannot find function, tuple struct or tuple variant `Some` in this scope
  --> C:\Users\pepe\.cargo\registry\src\github.com-1ecc6299db9ec823\anyhow-1.0.56\src\fmt.rs:35:43
   |
35 |                     number: if multiple { Some(n) } else { None },
   |                                           ^^^^ not found in this scope`

**Reply by mentasuave01** [reply]
well, rust was not correctly installed.
If someone having issues related to building could be the same issue, 
try to search on google for common mistakes installing rust in win

**Reply by parallelchainlab** [reply]
Thanks @mentasuave01 for resolving this issue. 

If you encounter any problems, please feel free to open a new discussion thread for us to easily track these questions.

If you encountered any major issues with the SDK, please file an issue at [https://github.com/parallelchain-io/parallelchain-sdk/issues](https://github.com/parallelchain-io/parallelchain-sdk/issues)

**Comment by mentasuave01** [comment]
```
Contract address: "3FCI7oThxiSkHJfM4wiQsWl2h1tthwoAsSjcngon4Lo="
Hash of tx: "thJDagYiyp9vT9MFlPk7H/ZrVbNOh7+eHN0UXku4lzs="
Status 200
```
Feel free to interact with my matcha shop!
Arguments: https://github.com/mentasuave01/Parallel-Chain-simple-contract-example/blob/master/src/lib.rs

**Reply by parallelchainlab** [reply]
Hi @mentasuave01 , thanks for using our sdk to develop your ParallelChain smart contract and share in here. Your contract code looks great without any issue. 
But please be aware that the output "_Status 200_" is a HTTP response status codes which implies your transaction request was successfully sent to our node to process. 
For your transaction status, you need to check the [Transaction Receipt](https://testnet.parallelchain.io/explorer/transactions/details?transaction_hash=thJDagYiyp9vT9MFlPk7H%2FZrVbNOh7%2BeHN0UXku4lzs=) to ensure you transaction is being executed successfully. Please refer to the docs ["Transaction Status Code"](https://docs.parallelchain.io/protocol/#status-code) section for details.

**Comment by parallelchainlab** [comment]
### Opening your own discussion thread.

Hi All,

To better streamline the discussion thread, we would like to introduce a few steps to opening your own discussion.

1. Go to [https://github.com/parallelchain-io/parallelchain-sdk/discussions](https://github.com/parallelchain-io/parallelchain-sdk/discussions)
2. Click on the button `New discussion` as shown in the image below.
![image](https://user-images.githubusercontent.com/95267993/161942731-1ec420c7-9df6-4abb-a523-8c0ba8a9cec5.png)
3. There are four categories to the discussion:
* General - the default setting for most users.
* Ideas - suggestions for us on our SDK. 
* Q&A - any problems you encounter when creating ParallelChain Smart Contract. We are an open source community. Feel free to help answer questions.
* Show and Tell - Any links to your smart contract source code, contract address and arguments.

4. Select the category of your discussion. In this example, we will be sharing our smart contract address and source code. The `category` that we used for the discussion is `Show and tell`. Furthermore, we will add links to our smart contract source code. 
![image](https://user-images.githubusercontent.com/95267993/161944628-fb934e9b-68c3-427c-9469-b4b5a8c9c471.png)

5. Click on `preview` to see how your discussion will appear. If you are satisfied with your post,  click on `Start discussion` to post your discussion
![image](https://user-images.githubusercontent.com/95267993/161944875-e10c5f27-f1e7-436c-a4f1-a86ab9110003.png)

6. Congratulations, your discussion is posted:
![image](https://user-images.githubusercontent.com/95267993/161945732-d225cd66-c4cc-4785-8918-3a85ad9b6dc4.png)


_Take note that all functions in the smart contract are public for now._

Thanks and Have fun.

**Reply by parallelchainlab** [reply]
This thread will be locked. Please refer to this guide above on how to make a new discussion thread.
